### DISCUSSION TITLE
cobin commits log selected check show wrong color

### DISCUSSION THREAD

**Discussion by woodgreat** [gitextensions/gitextensions#11084] [Q&A] [answered]
### Environment

- GitExtensions version: 3.5.4.12724
- GIT version: 2.41.0.windows.1
- OS version: windowns 10 NT 10.0.18363.0
- .NET version: 4.8.4341.0

combin commits log selected check show wrong color
I always use  our tool for git check work too,
but one day I fond ,when I select 1 commit ,it shown right info ,then I select muti-items it shown wrong color make me confustion. The color red & green one just inverted ,I means the add and remove items invert shown by git extensions : )

I got screen shot for you:

![2023-06-30_084002_correct](https://github.com/gitextensions/gitextensions/assets/38380995/68708bb2-df10-4320-b484-706b2c1b7d10)
![2023-06-30_084804_wrong](https://github.com/gitextensions/gitextensions/assets/38380995/449f7b7c-670c-4d0b-86a8-68660fb67749)

### Issue description

combin commits log selected check show wrong color
I always use  our tool for git check work too,
but one day I fond ,when I select 1 commit ,it shown right info ,then I select muti-items it shown wrong color make me confustion. The color red & green one just inverted ,I means the add and remove items invert shown by git extensions : )

### Steps to reproduce

just select many item same time you will see it happen !

### Did this work in previous version of GitExtensions?

not noticed

**Comment by vbjay** [comment, marked as answer]
The order matters. Select first commit and then second. Then start with second and then first.

**Reply by RussKie** [reply]
As @vbjay said, the order matters. Think of it this way - "_I want to see the difference between commits A and B_", and "_I want to see the difference between commits B and A_".

For example, I'm selecting the commits in different order:
![image](https://github.com/gitextensions/gitextensions/assets/4403806/e3baa45d-ea63-40b5-aa64-ea3635bddbfd)
![image](https://github.com/gitextensions/gitextensions/assets/4403806/881d3631-7841-4d4c-be30-ce6aab65cc2e)

**Comment by woodgreat** [comment]
the muti-selection matter is means :"I want see all these change want I made"

**Reply by mstv** [reply]
In order to see the changes done in these three selected commits,

![image](https://github.com/gitextensions/gitextensions/assets/36601201/53fedc3f-0dfc-41a1-a334-61d09c35179e)

- select the previous commit as diff base (just by clicking it)
- add the last of the commits you are interested in, to the selection (by ctrl+click)

![image](https://github.com/gitextensions/gitextensions/assets/36601201/e79bfafd-0b45-479f-918b-f44ba40fad01)

**Reply by woodgreat** [reply]
and then you will see the color is inverted as + and -

**Reply by woodgreat** [reply]
I saw it happen ,I wanna know why 2 differ came by differenct way I selected. from below and from top?why not define just 1 way for shown changed ? thank you .

**Reply by gerhardol** [reply]
Select first (normally oldest) then second commit. What is only in first is red, only in second green. 

You can multiselect commits for multi differences. If you select commits in separate branches you see range-diff with common commits.

**Reply by woodgreat** [reply]
I understood,

**Reply by woodgreat** [reply]
thank you
